### DISCUSSION TITLE
Any Way You Could Add a Gamepad Controller App from Your Achievements Sir Sergey?

### DISCUSSION THREAD

**Discussion by sevendemagnus** [community/community#110777] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Hi sir Sergey with your programming skills (like the Windows Install app to make life easier for Mac users), perhaps you could fix the errors in this app which makes it possible for macOS users (I believe) to make use of their gamepad controllers like the Sony Playstation DualSense so that you can use the gyroscope to control the Hang On motorcycle in Shenmue 1 using Redream (HD version) emulator. Shenmue is the father of all modern open-world games without it there'd be no GTA 3 to 5, nor Witcher 3, nor Cyberpunk 2077, nor Fallout 3 and 4.

This is the link to a most recommended program to complete. I only got to compile it a little but it has errors, I'm not a programmer perse. Perhaps you could complete it, it's almost 100% complete, it's by another programmer. You might want to try your hand on this:

https://github.com/marcowindt/ds4macos

On Windows there is a gamepad controller app caled DS4Windows it's very popular but this is not from the creator.

If it's ok inform me if you're successful in compiling it without errors.

Thank you. God bless you, have an amazing Lent and advanced Easter.

**Comment by queenofcorgis** [comment]
If you're asking how to use the piece of software that you found in a repository on GitHub or ask a question about it, the best way to contact the maintainers of that software is to:

- Check the README for instructions on how to operate the software and pointers to documentation or troubleshooting info.
- Check the [SUPPORT file](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-support-resources-to-your-project), if one exists, for instructions on how to best contact the maintainers for support.
- Check the [CONTRIBUTING guide](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors), if one exists. Sometimes if there isn't a SUPPORT file, the CONTRIBUTING guide will give instructions on how to contact the maintainers for support.

All of these documents can be found, if they exist, in the repository where you found the software itself.
